modules/base/rendering/pointcloud/renderablepolygoncloud.cpp

```diff
@@ -0,0 +1,194 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/pointcloud/renderablepolygoncloud.h>
+
+#include <openspace/documentation/documentation.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/glm.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/programobject.h>
+#include <ghoul/opengl/texture.h>
+#include <optional>
+
+namespace {
+    constexpr std::string_view _loggerCat = "RenderablePolygonCloud";
+
+    // A RenderablePolygonCloud is a RenderablePointCloud where the shape of the points
+    // is a uniform polygon with a given number of sides instead of a texture. For
+    // instance, PolygonSides = 5 results in the points being rendered as pentagons.
+    // Note that while this renderable inherits the texture property from
+    // RenderablePointCloud, any added texture value will be ignored in favor of the
+    // polygon shape.
+    //
+    // See documentation of RenderablePointCloud for details on the other parts of the
+    // point cloud rendering.
+    struct [[codegen::Dictionary(RenderablePolygonCloud)]] Parameters {
+        // The number of sides for the polygon used to represent each point. Default is
+        // 3, i.e. to use triangles
+        std::optional<int> polygonSides [[codegen::greaterequal(3)]];
+    };
+
+#include "renderablepolygoncloud_codegen.cpp"
+}  // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderablePolygonCloud::Documentation() {
+    return codegen::doc<Parameters>(
+        "base_renderablepolygoncloud",
+        RenderablePointCloud::Documentation()
+    );
+}
+
+RenderablePolygonCloud::RenderablePolygonCloud(const ghoul::Dictionary& dictionary)
+    : RenderablePointCloud(dictionary)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _polygonSides = p.polygonSides.value_or(_polygonSides);
+
+    // The texture to use for the rendering will be generated in initializeGl. Make sure
+    // we use it in the rnedering
+    _hasSpriteTexture = true;
+}
+
+void RenderablePolygonCloud::initializeGL() {
+    ZoneScoped;
+
+    RenderablePointCloud::initializeGL();
+    createPolygonTexture();
+}
+
+void RenderablePolygonCloud::deinitializeGL() {
+    glDeleteBuffers(1, &_polygonVbo);
+    _polygonVbo = 0;
+    glDeleteVertexArrays(1, &_polygonVao);
+    _polygonVao = 0;
+
+    glDeleteTextures(1, &_pTexture);
+}
```

Comment:
```suggestion

    RenderablePointCloud::deinitializeGL();
}
```

for symmetry with initialize